api/v1alpha1/acl_types.go

```diff
@@ -0,0 +1,25 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// +kubebuilder:validation:XValidation:rule="(has(self.allow) || has(self.deny))",message="one of allow or deny must be specified"
+//
+// ACL defines the IP deny/allow configuration.
+type ACL struct {
+	Allow []IPSpec `json:"allow,omitempty"`
+	Deny  []IPSpec `json:"deny,omitempty"`
+}
+
+// IPSpec defines the configuration for IP.
+type IPSpec struct {
```

Comment:
is there a common struct from upstream(gateway api or kubernetes)? maybe use `a.b.c.d/24` for better UX?